docs/byoh_agent.md

```diff
@@ -0,0 +1,116 @@
+# BYOH Agent
+
+BYOH agent is a binary that runs on the hosts and its responsibility include -  
+1. Registration - Register the host to the cluster capacity pool
+2. Setup - Install / Uninstall kubernetes components on the host
+3. Bootstrap - Convert the host to a kubernetes node using kubeadm
+
+## Usage of BYOH agent
+
+Below flags are supported by the agent:-  
+```
+--downloadpath string 
+```
+File System path to keep the downloads (default `/var/lib/byoh/bundles`)  
+<br />
+
+```
+--bootstrap-kubeconfig string           
+```
+Paths to a kubeconfig. Only required if out-of-cluster.  
+<br />
+```
+--label labelFlags       
+```
+labels to attach to the ByoHost CR in the form `labelname=labelVal` Eg: `--label site=apac --label cores=2`  
+<br />
+```
+--metricsbindaddress string
+```
+metricsbindaddress is the TCP address that the controller should bind to for serving prometheus metrics.It can be set to "0" to disable the metrics serving (default ":8080")  
+<br />
+```
+--namespace string
+```
+Namespace in the management cluster where you would like to register this host (default "default")  
+<br />
+```
+--skip-installation
+```
+If you want to skip installation of the kubernetes component binaries. If this flag is used, it will be user's responsibility to manage kubernetes components on the host.  
```

Comment:
```suggestion
If you want to skip the installation of the Kubernetes component binaries. If this flag is used, it will be the user's responsibility to manage Kubernetes components on the host.  
```